api/output_policy.go

```diff
@@ -0,0 +1,197 @@
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strings"
+
+	retryablehttp "github.com/hashicorp/go-retryablehttp"
+	"github.com/hashicorp/vault/sdk/helper/jsonutil"
+)
+
+const (
+	ErrOutputPolicyRequest = "output policy request"
+)
+
+var LastOutputPolicyError *OutputPolicyError
+
+type OutputPolicyError struct {
+	*retryablehttp.Request
+	VaultClient     *Client
+	parsingError    error
+	parsedHCLString string
+}
+
+func (d *OutputPolicyError) Error() string {
+	if d.parsedHCLString == "" {
+		d.parseRequest()
+		if d.parsingError != nil {
+			return d.parsingError.Error()
+		}
+	}
+
+	return ErrOutputPolicyRequest
+}
+
+// Builds a sample policy document from the request
+func (d *OutputPolicyError) parseRequest() {
```

Comment:
This function is called `parseRequest()` but it actually sneakily makes several requests. It may be worth splitting it up into a couple functions -- one for making the requests and one for accumulating capabilities. You may also want to rename the later to `buildSamplePolicy()` or something similar.

Reply:
Thanks for the rename idea!

Based on your other comment I think you may be misreading some of this, which I definitely don't blame you for, because the pattern taken by this and `-output-curl-string` is rather convoluted for sure. The method doesn't make several requests; it only makes one, to the OpenAPI endpoint, to determine the list of paths that require sudo.

Because of that, there didn't seem like a great way to split up the parseRequest method any further than it already is, but I took your naming advice and renamed the whole method to `buildSamplePolicy` (and something similar for -output-curl-string), and also changed a few of the other properties to make their purpose hopefully a little more clear.

Now that I've changed those the whole thing is a little more readable now in my opinion. So thank you!